cql3/expr/expression.cc

```diff
@@ -103,7 +103,13 @@ managed_bytes_opt get_value(const column_value& col, const column_value_eval_bag
         if (!col_type->is_map()) {
             throw exceptions::invalid_request_exception(format("subscripting non-map column {}", cdef->name_as_text()));
         }
-        const auto deserialized = cdef->type->deserialize(managed_bytes_view(*data.other_columns[data.sel.index_of(*cdef)]));
+        int32_t index = data.sel.index_of(*cdef);
+        if (index == -1) {
+            throw exceptions::invalid_request_exception(
+                    format("Column definition {} does not match between base and view tables",
+                    cdef->name_as_text()));
```

Comment:
I don't understand why this error is specific to "match between base and view tables". Maybe some materialized-view code caller may interpret this exception as what you're saying here, but I don't understand how this function can infer this interpretation here. 

You said correctly that this patch will help you debug some materialized-view-related bug you're seeing. However, I'm worried that this specific wording will in the future give us problems when debugging a bug that isn't actually related to materialized views but will print messages blaming materialized views.

Reply:
+1
I think the error message should be such that we find the place in the code where it was thrown from easily and provides as much information as reasonable, but doesn't try to interpret the error condition.

BTW we could also use `at()` instead of `operator[]` in these spots to achieve the same effect, but `at()` has a very unspecific error message.

Reply:
Ah, sure, I forgot that this code isn't view-specific